torch/nested/_internal/ops.py

```diff
@@ -713,7 +716,35 @@ def transpose_int(func, *args, **kwargs):
         func, args=args, kwargs=kwargs, normalize_to_only_use_kwargs=True
     )
 
+    from torch._prims_common import canonicalize_dims
+
     inp = new_kwargs.pop("input")
+    dim0, dim1 = canonicalize_dims(inp.dim(), (new_kwargs["dim0"], new_kwargs["dim1"]))
+
+    # To support the SDPA API, inputs need to have the ragged idx transposed to dim 2
+    # instead of 1, although the internal Flash and mem-effn implementations will
+    # use the inputs with raggedness in dim 1. Therefore, when requesting this transpose,
+    # move things around but don't change anything in _values
+    if dim0 == inp._ragged_idx or dim1 == inp._ragged_idx:
+        if dim0 == 0 or dim1 == 0:
+            raise ValueError(
+                "Transpose is not supported on the batch dimension for jagged NT"
+            )
+        if dim0 == inp._ragged_idx:
+            to_dim = dim1
+        else:
+            to_dim = dim0
+        inp._ragged_idx = to_dim
+        new_size = list(inp._size)
+        new_size[dim0], new_size[dim1] = new_size[dim1], new_size[dim0]
+        inp._size = tuple(new_size)
+        inp._values.transpose_(
```

Comment:
I'm (probably irrationally) scared of in-place views like this :p

is this correct? I actually think we want to return a new `NestedTensor` with transposed `values` and not modify anything in `inp`. note that this is `transpose.int`, which is an out-of-place operation

Reply:
changed to not use inp as a temp value storage